cmake/configure/configure_1_mpi.cmake

```diff
@@ -17,10 +17,45 @@
 # Configuration for mpi support:
 #
 
+MACRO(FEATURE_MPI_FIND_EXTERNAL var)
+  FIND_PACKAGE(MPI)
+
+  IF(MPI_FOUND)
+    SET(${var} TRUE)
+    #
+    # Older versions of MPI may not have MPI_SEEK_SET, which we
+    # require. Strangely, unlike MPICH, OpenMPI needs the correct link libraries
+    # for this to compile, not *just* the correct include directories.
+    #
+    CLEAR_CMAKE_REQUIRED()
+    SET(CMAKE_REQUIRED_FLAGS ${MPI_LINKER_FLAGS})
+    SET(CMAKE_REQUIRED_INCLUDES ${MPI_INCLUDE_DIRS})
+    SET(CMAKE_REQUIRED_LIBRARIES ${MPI_LIBRARIES})
+    CHECK_CXX_SOURCE_COMPILES("
```

Comment:
May I ask you to indent appropriately
```
CMAKE_CXX_SOURCE_COMPILES(
  "
  // source code
  "
  DEAL_II_HAVE_MPI_SEEK_SET)
```

and refactor the ```CHECK_CXX_SOURCE_COMPILES``` to ```modules/FindMPI.cmake```, update the comment on what ```FindMPI.cmake``` defines and also put ```DEAL_II_HAVE_MPI_SEEK_SET``` into the ```CLEAR``` part of ```DEAL_II_CONFIGURE_FEATURE```? (Have a look at ```FindTRILINOS.cmake```, etc.).